### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by breriksenms**
#### Pull Request Description

Adding a new sample for how to use the sdk that includes some advanced querying techniques, like query parameters and extracting individual values from the results.

### CODE REVIEW

File: samples/src/main/java/AdvancedQuery.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            HttpClientProperties properties = HttpClientProperties.builder()
+                    .keepAlive(true)
+                    .maxKeepAliveTime(120)
+                    .maxConnectionsPerRoute(40)
+                    .maxConnectionsTotal(40)
+                    .build();
+
+            ClientImpl client = new ClientImpl(csb, properties);
+            String database = System.getProperty("dbName");
+            String newLine = System.getProperty("line.separator");
```
Comment (ohadbitt) [inline samples/src/main/java/AdvancedQuery.java:34]: why needed

**Review by ohadbitt — CHANGES_REQUESTED** [review]

File: samples/src/main/java/AdvancedQuery.java (hunk 1 above)
Comment (breriksenms) [reply samples/src/main/java/AdvancedQuery.java:34]: It's used to separate the lines of the query below for readability. Would you prefer to use a different character instead?

File: samples/README.md [hunk 2]
```diff
@@ -72,6 +72,68 @@ mvn clean compile exec:java -Dexec.mainClass="Query" \
...
+
+### How to run this sample
+Note: Running this sample will create a table named `Events` in the given database.
+```sh
+cd samples
+mvn clean compile exec:java -Dexec.mainClass="AdvancedQuery" \
+                            -DclusterPath="cluster/path" \
+                            -DappId="app-id" \
+                            -DappKey="appKey" \
+                            -DappTenant="subscription-id" \
```
[the remark this answers is not in the document]
Comment (breriksenms) [reply samples/README.md]: updated to tenant-id

File: samples/README.md [hunk 3]
```diff
@@ -72,6 +72,68 @@ mvn clean compile exec:java -Dexec.mainClass="Query" \
...
+                            -DdbName="dbName" 
+```
+
 ## File Ingestion Sample
 
 This sample will demonstrate how to ingest data from file into table.
```
[the remark this answers is not in the document]
Comment (breriksenms) [reply samples/README.md]: updated

File: samples/src/main/java/AdvancedQuery.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            String query = String.join(newLine,
+                    "declare query_parameters(eventNameFilter:string);",
+                    "Events",
+                    "| where eventName == eventNameFilter");
+            KustoOperationResult results = client.execute(database, query, clientRequestProperties);
+            KustoResultSetTable mainTableResult = results.getPrimaryResults();
+            System.out.printf("Kusto sent back %s rows.%n", mainTableResult.count());
+
+            // iterate values
+            List<Event> events = new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (breriksenms) [reply samples/src/main/java/AdvancedQuery.java:58]: Could you point me towards the pojo?

File: samples/src/main/java/AdvancedQuery.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            clientRequestProperties.setParameter("eventNameFilter", "event 1");
+            String query = String.join(newLine,
+                    "declare query_parameters(eventNameFilter:string);",
+                    "Events",
+                    "| where eventName == eventNameFilter");
+            KustoOperationResult results = client.execute(database, query, clientRequestProperties);
+            KustoResultSetTable mainTableResult = results.getPrimaryResults();
+            System.out.printf("Kusto sent back %s rows.%n", mainTableResult.count());
+
+            // iterate values
```
[the remark this answers is not in the document]
Comment (breriksenms) [reply samples/src/main/java/AdvancedQuery.java]: Updated

File: samples/README.md [hunk 6]
```diff
@@ -72,6 +72,68 @@ mvn clean compile exec:java -Dexec.mainClass="Query" \
...
 
+## Advanced Query Sample
+
+This sample shows some more advanced options available when querying data, like using query parameters to guard against injection attacks and extracting individual values from the query results.   
+[Sample Code](src/main/java/Query.java)
+
+### Prerequisites
+
+- [Create Azure Data Explorer Cluster and DB](https://docs.microsoft.com/en-us/azure/data-explorer/create-cluster-database-portal)
+- [Create Azure Active Directory App Registration and grant it permissions to DB](https://docs.microsoft.com/en-us/azure/kusto/management/access-control/how-to-provision-aad-app) (save the app key and the application ID for later). Permissions must include the ability to create tables in the database.
```
[the remark this answers is not in the document]
Comment (breriksenms) [reply samples/README.md]: Updated

File: samples/README.md [hunk 7]
```diff
@@ -72,6 +72,68 @@ mvn clean compile exec:java -Dexec.mainClass="Query" \
...
 ```
 
+## Advanced Query Sample
+
+This sample shows some more advanced options available when querying data, like using query parameters to guard against injection attacks and extracting individual values from the query results.   
+[Sample Code](src/main/java/Query.java)
+
+### Prerequisites
+
+- [Create Azure Data Explorer Cluster and DB](https://docs.microsoft.com/en-us/azure/data-explorer/create-cluster-database-portal)
```
[the remark this answers is not in the document]
Comment (breriksenms) [reply samples/README.md]: Updated.

File: samples/src/main/java/AdvancedQuery.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (ohadbitt) [reply samples/src/main/java/AdvancedQuery.java:58]: just said consider - you dont have to 
https://github.com/Azure/azure-kusto-java/blob/ca04a5881ac52c1861775611a505cadf6cfa2308/data/src/test/java/com/microsoft/azure/kusto/data/KustoResultMapperTest.java

File: samples/src/main/java/AdvancedQuery.java (hunk 1 above)
Comment (ohadbitt) [reply samples/src/main/java/AdvancedQuery.java:34]: just asked if there was a reason for the way you acquire it

**Pull request merged**